src/Widgets/SharePopover.vala

```diff
@@ -0,0 +1,165 @@
+/*-
+ * Copyright (c) 2017 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Daniel Foré <daniel@elementary.io>
+ */
+
+public class SharePopover : Gtk.Popover {
+    public signal void link_copied ();
+
+    public string body { get; set; }
+    public string uri { get; set; }
+
+    public SharePopover (string body, string uri) {
+        Object (
+            body: body,
+            uri: uri
+        );
+    }
+
+    construct {
+        var email_button = new Gtk.Button.from_icon_name ("internet-mail", Gtk.IconSize.DND);
+        email_button.tooltip_text = _("Email");
+        email_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+
+        var facebook_button = new Gtk.Button.from_icon_name ("online-account-facebook", Gtk.IconSize.DND);
+        facebook_button.tooltip_text = _("Facebook");
+        facebook_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+
+        var google_button = new Gtk.Button.from_icon_name ("online-account-google-plus", Gtk.IconSize.DND);
+        google_button.tooltip_text = _("Google+");
+        google_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+
+        var twitter_button = new Gtk.Button.from_icon_name ("online-account-twitter", Gtk.IconSize.DND);
+        twitter_button.tooltip_text = _("Twitter");
+        twitter_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+
+        var reddit_button = new Gtk.Button.from_icon_name ("online-account-reddit", Gtk.IconSize.DND);
+        reddit_button.tooltip_text = _("reddit");
+        reddit_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+
+        var tumblr_button = new Gtk.Button.from_icon_name ("online-account-tumblr", Gtk.IconSize.DND);
+        tumblr_button.tooltip_text = _("Tumblr");
+        tumblr_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+
+        var pocket_button = new Gtk.Button.from_icon_name ("online-account-pocket", Gtk.IconSize.DND);
+        pocket_button.tooltip_text = _("Save to Pocket");
```

Comment:
It's literally an article saving service. Apps are not articles.

![screenshot from 2017-06-29 20 07 56](https://user-images.githubusercontent.com/611168/27718103-dc4bfa04-5d06-11e7-878f-efab690e827c.png)

![screenshot from 2017-06-29 20 08 32](https://user-images.githubusercontent.com/611168/27718104-dc4c0562-5d06-11e7-8fd7-992cb03ed361.png)